spec/independent_styles_spec.rb

```diff
@@ -1,79 +1,76 @@
-Independents.each_pair do |id, (filename, path, style, reason)|
+Independents.each_pair do |basename, (filename, path, style, reason)|
 
-  describe "independent style #{id}" do
+  describe "#{basename}:" do
 
-    it "is a valid CSL 1.0 style" do
+    it "must validate against the CSL 1.0.1 schema" do
       expect(CSL.validate(path)).to eq([])
     end
 
-    it "has a conventional file name" do
+    it "must have a conventional file name" do
       expect(filename).to match(/^[a-z\d]+(-[a-z\d]+)*\.csl$/)
     end
 
-    it "was successfully parsed" do
+    it "must be parsable as a CSL style" do
       expect(style).to be_a(CSL::Style), reason
     end
 
     unless style.nil?
-      it "is independent" do
+      it "only independent styles are allowed in the root directory" do
         expect(style).to be_independent
       end
 
-      it "has an info element" do
+      it "must contain an <info/> element" do
        expect(style).to have_info
       end
 
-      it "has a self-link" do
+      it 'must have a "self" link' do
         expect(style).to have_self_link
       end
 
-      it "has an id" do
+      it "must have a style ID" do
         expect(style.info).to have_id
       end
 
-      it "the id is a valid style repository link" do
-        expect(style.id).to eq("http://www.zotero.org/styles/#{id}")
+      it 'style ID must be of the form "http://www.zotero.org/styles/" + style file name (without ".csl" extension, e.g. "http://www.zotero.org/styles/apa")' do
+        expect(style.id).to eq("http://www.zotero.org/styles/#{basename}")
       end
 
-      it "the self-link is a valid style repository link" do
-        expect(style.self_link).to match(%r{http[s]?://www.zotero.org/styles/#{id}})
+      it '"self" link must match the style ID' do
+        if style.has_self_link?
+          expect(style.id).to eq(style.self_link)
+        end
       end
 
-      it "has an info/rights element" do
+      it "must have a <rights> element" do
         expect(style.info).to have_rights
       end
 
-      it "is licensed under a CC BY-SA license" do
+      it "must have the correct Creative Commons BY-SA license" do
         expect(style).to be_default_license
       end
 
-      it "its template-link (if present) points to an existing independent style" do
+      it '"template" link must point to an existing independent style' do
         if style.has_template_link?
-          link = style.template_link
-
-          expect(link).to match(%r{http[s]?://www.zotero.org/styles/([a-z-]+)})
-          expect(Independents).to have_key(link[/[^\/]+$/])
+          parent_ID_link = style.template_link
+          link_prefix = "http://www.zotero.org/styles/"
+          
+          expect(parent_ID_link).to match(%r{^#{link_prefix}})
+          parent_ID = parent_ID_link[link_prefix.length..-1]
```

Comment:
The `independent?` predicate is [defined by the presence/absence of the link](https://github.com/inukshuk/csl-ruby/blob/master/lib/csl/style.rb#L108) so this is being tested implicitly. It may be helpful to add an explicit test with a more meaningful error message, or to change the 'must be independent' to 'may not contain a link to an independent parent'.